src/services/Api.service.ts

```diff
@@ -0,0 +1,39 @@
+import { toast } from 'react-toastify';
+
+import type { AxiosResponse } from 'axios';
+import axios from 'axios';
+
+export class ApiService {
+    private static INSTANCE: ApiService;
+    private baseURL: string = 'https://ma-backend-api.mocintra.com/api/v1';
+
+    private constructor() {
+        axios.defaults.baseURL = this.baseURL;
+        axios.interceptors.response.use(
+            (response) => response,
+            (error) => {
+                toast.error(error.message);
+                return Promise.reject(error);
+            },
+        );
+    }
+
+    public static GetInstance(): ApiService {
```

Comment:
Cool OOP solution with a static method.
Another option: you can export not the class itself, but the created example and use it as a singleton in whole app:

Reply:
fixed